src/lightning/pytorch/core/module.py

```diff
@@ -1539,20 +1538,9 @@ def load_from_checkpoint(
         )
         return cast(Self, loaded)
 
-    @contextmanager
-    def _prevent_trainer_and_dataloaders_deepcopy(self) -> Generator[None, None, None]:
-        self._should_prevent_trainer_and_dataloaders_deepcopy = True
-        yield
-        self._should_prevent_trainer_and_dataloaders_deepcopy = False
-
     def __getstate__(self) -> Dict[str, Any]:
         state = dict(self.__dict__)
-        if self._should_prevent_trainer_and_dataloaders_deepcopy:
-            state["_trainer"] = None
-            state.pop("train_dataloader", None)
-            state.pop("val_dataloader", None)
-            state.pop("test_dataloader", None)
-            state.pop("predict_dataloader", None)
```

Comment:
These are not needed anymore: https://github.com/Lightning-AI/lightning/pull/8472#pullrequestreview-710817996